WalletWasabi/Services/SystemAwakeChecker.cs

```diff
@@ -18,17 +19,17 @@ public class SystemAwakeChecker : PeriodicRunner
 
 		private volatile IPowerSavingInhibitorTask? _powerSavingTask;
 
-		private SystemAwakeChecker(WalletManager walletManager, Func<Task<IPowerSavingInhibitorTask>>? taskFactory) : base(TimeSpan.FromMinutes(1))
+		private SystemAwakeChecker(CoinJoinManager coinJoinManager, Func<Task<IPowerSavingInhibitorTask>>? taskFactory) : base(TimeSpan.FromSeconds(5))
```

Comment:
This is very heavy weight. Adding tests for `SystemAwakeChecker` would be harder. Maybe we can just change it to something like:

`private SystemAwakeChecker(Func<CoinJoinClientState> stateFunction, Func<Task<IPowerSavingInhibitorTask>>? taskFactory) : base(TimeSpan.FromSeconds(5))`

Then we can really pass anything in tests to `SystemAwakeChecker`. If `Func<CoinJoinClientState>` is too long, one can define an alias for that using `using` in C#.

Reply:
I am against this but SystemAwakeChecker refactoring will be done in the next PR.

Reply:
Agains `Func<CoinJoinClientState>` idea? Or passing `CoinJoinManager coinJoinManager` to `SystemAwakeChecker`?

Reply:
Against passing Func. I prefer Interfaces.